query_executor.go

```diff
@@ -48,16 +48,64 @@ type ExecutableQuery interface {
 }
 
 type queryExecutor struct {
-	pool   *policyConnPool
-	policy HostSelectionPolicy
+	pool        *policyConnPool
+	policy      HostSelectionPolicy
+	interceptor QueryAttemptInterceptor
+}
+
+type QueryAttempt struct {
+	// The query to execute, either a *gocql.Query or *gocql.Batch.
+	Query ExecutableQuery
+	// The connection used to execute the query.
+	Conn *Conn
+	// The host that will receive the query.
+	Host *HostInfo
```

Comment:
Do we need to add and expose these fields in `QueryAttempt`? 

If there’s no compelling need for direct connection manipulation, then maybe we could expose only a connection metadata that’s relevant for logging or monitoring purposes?

Reply:
Removed the `Conn` field in favor of read-only `LocalAddr` and `RemoteAddr` connection metadata fields, but retained the `Host` field to match `ObservedQuery`.